flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/assigners/splitters/SampleBucketSplitStrategy.java

```diff
@@ -105,9 +108,18 @@ public Collection<SnapshotSplit> split(SplitContext splitContext) {
             pipeline.add(sample(numberOfSamples));
         }
```

Comment:
```java
        if (count < DEFAULT_SAMPLING_THRESHOLD) {
            // full sampling if document count less than sampling size threshold.
            numberOfSamples = (int) count;
        } else {
            // sampled using sample rate.
            numberOfSamples = (int) Math.floor(count * DEFAULT_SAMPLING_RATE);
        }
```
As the old code always `numberOfSamples <= count`.
If the user configuration is open,  there's the possibility of `numberOfSamples > count` .

So, I think `if (numberOfSamples < count)` is safe here.